#! /bin/sh /usr/share/dpatch/dpatch-run
## 60_all_keyword_start_only.patch by Kel Modderman <kel@otaku42.de>
##
## DP: Purpose: Prevent Required-Stop: $all from modifying start links
## DP: Fixes:   #485307
## DP: Status:  Work in progress.

@DPATCH@
--- a/insserv.c
+++ b/insserv.c
@@ -291,8 +291,13 @@
 	    break;
 	case '$':
 	    if (strcasecmp(token, "$all") == 0) {
-		serv->attr.flags |= SERV_ALL;
-		break;
+		if (bit & REQ_KILL) {
+		    /* $all has no effect on stop sort order */
+		    break;
+		} else {
+		    serv->attr.flags |= SERV_ALL;
+		    break;
+		}
 	    }
 	    /* Expand the `$' token recursively down */
 	    list_for_each(ptr, sysfaci_start) {
